### PR TITLE
Feat: Add save to local storage

### LINKED ISSUE #11: New User Creation Method
As an User I would like to have my information saved in the local storage after filling the New User Form and clicking on the Create User Button

### PR DESCRIPTION
**Pull request by ivanvgav**
Added function to save user data into local storage.

### CODE REVIEW

File: js/newUser.js [hunk 1]
```diff
@@ -20,10 +20,43 @@ form.addEventListener('submit', (e) => {
...
-        }
+    }
     catch (error) {
         alert("You made a mistake. Please try again.");
         console.log(error);
     }
-    })
+})
+
+function saveToLocalStorage() {
```
Comment (gdeamicis) [inline js/newUser.js]:
```suggestion
function saveToLocalStorage(email, username, password) {
  const user = { email, username, encryptedPassword: Hash(password) }
```

File: js/newUser.js [hunk 2]
```diff
@@ -20,10 +20,43 @@ form.addEventListener('submit', (e) => {
         const password = document.getElementById('password').value;
         alert(`User create succesfully. The date is Username: ${username}, Email: ${email} and Password: ${password}`);
 
+        saveToLocalStorage()
```
Comment (gdeamicis) [inline js/newUser.js]:
```suggestion
        saveToLocalStorage(email, username, password)
```

File: js/newUser.js [hunk 3]
```diff
@@ -20,10 +20,43 @@ form.addEventListener('submit', (e) => {
...
         alert("You made a mistake. Please try again.");
         console.log(error);
     }
-    })
+})
+
+function saveToLocalStorage() {
+    let newEmail = document.getElementById('email').value
+    let newUsername = document.getElementById('username').value
+    let newPassword = document.getElementById('password').value
```
Comment (gdeamicis) [inline js/newUser.js]: not necessary

File: js/newUser.js [hunk 4]
```diff
@@ -20,10 +20,43 @@ form.addEventListener('submit', (e) => {
...
+
+    if (localStorage.getItem('email') == null) {
+        localStorage.setItem('email', '[]' )
```
Comment (gdeamicis) [inline js/newUser.js]:
```suggestion
        localStorage.setItem('email', JSON.stringify([account]) )
```

File: js/newUser.js [hunk 5]
```diff
@@ -20,10 +20,43 @@ form.addEventListener('submit', (e) => {
...
+    }
```
Comment (gdeamicis) [inline js/newUser.js]:
```suggestion
    } else {
      let users = JSON.parse(localStorage.getItem('users'))
      users.push(account)
      localStorage.setItem('users', JSON.stringify(users));
      }
```

File: js/newUser.js [hunk 6]
```diff
@@ -20,10 +20,43 @@ form.addEventListener('submit', (e) => {
...
+
+    let oldEmail = JSON.parse(localStorage.getItem('email'))
+    oldEmail.push(newEmail)
+
+    let oldUsername = JSON.parse(localStorage.getItem('username'))
+    oldUsername.push(newUsername)
+
+    let oldPassword = JSON.parse(localStorage.getItem('password'))
+    oldPassword.push(newPassword)
```
Comment (gdeamicis) [inline js/newUser.js]: NEVER SAVE PASSWORD IN PLAIN TEXT

Comment (gdeamicis) [reply js/newUser.js]: look for encryption algorithms to save the encrypted password in local storage. The encryption algorithm should be 1:1.

**Comment by ivanvgav** [comment]
I fought a lot of trying to import the library to encrypt the password. A lot. And it was a simple cdn, but the docs of the library are not clear to do that.

I had the following problems:

1. There is an error in the encryptPassword function that cannot make to encrypt the password. I think that is with the password value.
2. I do not know where to set the object users and then set the possibles individuals users.

File: js/newUser.js [hunk 7]
```diff
@@ -19,11 +19,77 @@ form.addEventListener('submit', (e) => {
         const username = document.getElementById('username').value;
         const password = document.getElementById('password').value;
         alert(`User create succesfully. The date is Username: ${username}, Email: ${email} and Password: ${password}`);
+        
+        let users = {
```
Comment (ivanvgav) [inline js/newUser.js]: I do not know if it fine here or I have to set it in saveLocalStorage function

File: js/newUser.js [hunk 8]
```diff
@@ -19,11 +19,77 @@ form.addEventListener('submit', (e) => {
...
+            email: email,
+            username: username,
+            password: encryptPassword(password),
+        }
+
+        saveToLocalStorage(users)
```
Comment (ivanvgav) [inline js/newUser.js]: I think that the parameter is not right

File: js/newUser.js [hunk 9]
```diff
@@ -19,11 +19,77 @@ form.addEventListener('submit', (e) => {
...
+    if (user == null ) {
+        user = {
+            email: [],
+            username: [],
+            password: [],
+        }
+    } else {
+        let users = JSON.parse(localStorage.getItem('users'));
+        users.push(users);
+        localStorage.setItem('users', JSON.stringify(users));
```
Comment (ivanvgav) [inline js/newUser.js]: The push method is wrong. But I do not know what variable use to fix it.

**Pull request merged**